underfs/s3/src/main/java/tachyon/underfs/s3/S3InputStream.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Licensed to the University of California, Berkeley under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.underfs.s3;
+
+import java.io.BufferedInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+
+import org.jets3t.service.S3Service;
+import org.jets3t.service.ServiceException;
+import org.jets3t.service.model.S3Object;
+
+/**
+ * This class is a wrapper around the input stream returned by
+ * {@link org.jets3t.service.model.S3Object} getDataInputStream. The main purpose is to provide a
+ * faster skip method, as the underlying implementation will read and discard bytes until the
+ * number to skip has been reached.
+ */
+public class S3InputStream extends InputStream {
+
+  /** Bucket name of the Tachyon S3 bucket */
+  private final String mBucketName;
+  /** Key of the file in S3 to read */
+  private final String mKey;
+  /** The JetS3t client for S3 operations */
+  private final S3Service mClient;
+
+  /** The storage object that will be updated on each large skip */
+  private S3Object mObject;
+  /** The underlying input stream */
+  private BufferedInputStream mInputStream;
+  /** Position of the stream */
```

Comment:
This is the position of the exposed stream and not the position of the internal/underlying stream, right?